app/src/main/java/org/kiwix/kiwixmobile/zimManager/Fat32Checker.kt

```diff
@@ -63,11 +64,20 @@ class Fat32Checker constructor(
   }
 
   private fun fileObserver(it: String): FileObserver =
-    object : FileObserver(File(it), MOVED_FROM or DELETE) {
-      override fun onEvent(event: Int, path: String?) {
-        requestCheckSystemFileType.onNext(Unit)
-      }
-    }.apply { startWatching() }
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+      object : FileObserver(File(it), MOVED_FROM or DELETE) {
+        override fun onEvent(event: Int, path: String?) {
+          requestCheckSystemFileType.onNext(Unit)
+        }
+      }.apply { startWatching() }
+    } else {
+      @Suppress("DEPRECATION")
+      object : FileObserver(it, MOVED_FROM or DELETE) {
+        override fun onEvent(event: Int, path: String?) {
+          requestCheckSystemFileType.onNext(Unit)
```

Comment:
@MohitMaliFtechiz We may need in other location to do this file observe.We can't repeat this forever. I've made some changes with `FileObserverWrapper` class. Let me know if this works better in future.
```kotlin
private fun createFileObserver(it: String): FileObserver {
    val fileObserver = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
        FileObserver(File(it), MOVED_FROM or DELETE)
    } else {
        @Suppress("DEPRECATION")
        FileObserver(it, MOVED_FROM or DELETE)
    }

    return object : FileObserverWrapper(fileObserver) {
        override fun onEvent(event: Int, path: String?) {
            requestCheckSystemFileType.onNext(Unit)
        }
    }.apply { startWatching() }
}

abstract class FileObserverWrapper(private val fileObserver: FileObserver) : FileObserver(fileObserver.path, fileObserver.mask) {
    override fun onEvent(event: Int, path: String?) {
        fileObserver.onEvent(event, path)
    }
}
```

Reply:
@gouri-panda Is this code working? `FileObserver` is an abstract class we can not use this class like this, and there is no `path` and `mask` available in `fileObserver` class.

![Screenshot from 2024-01-11 17-46-20](https://github.com/kiwix/kiwix-android/assets/34593983/94cfcc16-d8f7-4a99-acb5-856960816c5f)

Reply:
It was an abstract idea. Do the architect work 🤔?

Reply:
Your idea is good to do it like this, Can you put the working code example of what you are exactly thinking?

> We can put the string in the path and mask object.

 This constructor is deprecated and may be not work properly on the latest versions of Android.
```
@Deprecated
    public FileObserver(String path, @NotifyEventType int mask) {
        this(new File(path), mask);
    }
```

Reply:
@MohitMaliFtechiz Sorry! I thought `path` and `mast` variables are open. Ok, we can go with the previous approach.